Microsoft.DependencyInjection.SolrNet.Tests/MicrosoftDependencyFixture.cs

```diff
@@ -179,8 +212,8 @@ public void SetBasicAuthenticationHeaderForTypedInstance()
             //test
             var provider = sc.BuildServiceProvider();
             var injectionConnection = provider.GetRequiredService<ISolrInjectedConnection<string>>() as BasicInjectionConnection<string>;
-            var connection = injectionConnection.Connection as AutoSolrConnection;
-            Assert.NotNull(connection.HttpClient.DefaultRequestHeaders.Authorization);
+            var connection = injectionConnection?.Connection as AutoSolrConnection;
+            Assert.NotNull(connection?.HttpClient.DefaultRequestHeaders.Authorization);
```

Comment:
Why was the reason to the null-conditional operators ` ?.`  ? 
Because this changes the test completely, in the past `Assert.NotNull`  would only check whether the `Authorization` property is null. If any of the other properties would have been null an exception would have been thrown (which is fine).
Now `Assert.NotNull` checks for null in `injectionConnection` , `injectionConnection.Connection`, and `injectionConnection.Connection.HttpClient.DefaultRequestHeaders.Authorization`.

Reply:
I added this to make ReSharper happy. I can revert this change. The outcome of the test is the same since all of those values have to be specified for the test to pass anyway.

Reply:
Reverted.

Reply:
Thanks! I understand that the outcome is the same, however we want to make sure we are validating the right thing, in case something changes.